rotkehlchen/db/snapshots.py

```diff
@@ -223,27 +226,33 @@ def update(
             location_data_snapshot: List[LocationData],
     ) -> Tuple[bool, str]:
         """Updates a DB Balance snapshot at a given timestamp."""
-        # delete the existing snapshot of that timestamp
-        is_success, message = self.delete(timestamp)
-        if is_success is False:
-            return is_success, message
+        with self.db.user_write() as cursor:
```

Comment:
What are you doing here? This is wrong. 

The entire point of using the write cursor is automatic rollback in the case of an exception.

But `delete()` does not raise an exception. So what's the point of this? 

For example if the `cursor.execute('DELETE FROM timed_balances WHERE time=?', (timestamp,))` works and `cursor.execute('DELETE FROM timed_location_data WHERE time=?', (timestamp,))` you should rollback.

But currently you will just return False, and get out of the context manager without rolling back.